src/structures/idt.rs

```diff
@@ -779,34 +779,47 @@ impl<F> Entry<F> {
     }
 }
 
-macro_rules! impl_set_handler_fn {
-    ($h:ty) => {
+#[cfg(all(feature = "instructions", feature = "abi_x86_interrupt"))]
+impl<F: HandlerFuncType> Entry<F> {
+    /// Set the handler function for the IDT entry and sets the present bit.
+    ///
+    /// For the code selector field, this function uses the code segment selector currently
+    /// active in the CPU.
+    ///
+    /// The function returns a mutable reference to the entry's options that allows
+    /// further customization.
+    ///
+    /// This method is only usable with the `abi_x86_interrupt` feature enabled. Without it, the
+    /// unsafe [`Entry::set_handler_addr`] method has to be used instead.
+    #[inline]
+    pub fn set_handler_fn(&mut self, handler: &F) -> &mut EntryOptions {
+        unsafe { self.set_handler_addr(handler.to_handler_fn_addr()) }
+    }
+}
+
+/// The trait for all function types that can be used as a handler function in `Entry`.
+pub trait HandlerFuncType {
+    /// Get the virtual memory address of the function
+    fn to_handler_fn_addr(&self) -> VirtAddr;
+}
+
+macro_rules! impl_handler_func_type {
+    ($f:ty) => {
         #[cfg(all(feature = "instructions", feature = "abi_x86_interrupt"))]
-        impl Entry<$h> {
-            /// Set the handler function for the IDT entry and sets the present bit.
-            ///
-            /// For the code selector field, this function uses the code segment selector currently
-            /// active in the CPU.
-            ///
-            /// The function returns a mutable reference to the entry's options that allows
-            /// further customization.
-            ///
-            /// This method is only usable with the `abi_x86_interrupt` feature enabled. Without it, the
-            /// unsafe [`Entry::set_handler_addr`] method has to be used instead.
+        impl HandlerFuncType for $f {
             #[inline]
-            pub fn set_handler_fn(&mut self, handler: $h) -> &mut EntryOptions {
-                let handler = VirtAddr::new(handler as u64);
-                unsafe { self.set_handler_addr(handler) }
+            fn to_handler_fn_addr(&self) -> VirtAddr {
```

Comment:
This should probably be a by-value method, and we could name it something like `as_virt_addr` for consistency with other type conversion methods.

Reply:
Apparently according to https://rust-lang.github.io/rust-clippy/master/index.html#/wrong_self_convention and https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv the convention would be to call this `into_virt_addr` or `to_virt_addr`.

I think `into_virt_addr` is fine.

Reply:
Whoops, rereading the guidelines again, it seems like `to_virt_addr` is what should be used for copy types, even if the conversion is cheap. For example, [`i32::to_ne_bytes()` ](https://doc.rust-lang.org/nightly/std/primitive.i32.html#method.to_ne_bytes)